src/baseline.rs

```diff
@@ -340,3 +363,113 @@ fn need_retry(res: Result<(), crates_index::Error>) -> anyhow::Result<bool> {
         Err(err) => Err(err.into()),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crates_index::{Crate, Version};
+
+    use super::choose_baseline_version;
+
+    fn new_mock_version(version_name: &str, yanked: bool) -> Version {
+        // `crates_index::Version` cannot be created explicitly, as all its fields
+        // are private, so we use the fact that it can be deserialized.
+        serde_json::from_value(serde_json::json!({
+            "name": "test-crate",
+            "vers": version_name,
+            "deps": [],
+            "features": {},
+            "yanked": yanked,
+            "cksum": "00".repeat(32),
+        }))
+        .expect("Failed to create crates_index::Version from a hand-written JSON.")
+    }
+
+    fn new_crate(versions: Vec<Version>) -> Crate {
+        // `crates_index::Crate` cannot be created explicitly, as its field
+        // is private, so we use the fact that it can be deserialized.
+        serde_json::from_value(serde_json::json!({ "versions": versions }))
+            .expect("Failed to create crates_index::Crate from a hand-written JSON.")
+    }
+
+    fn assert_correctly_picks_baseline_version(
+        versions: Vec<(&str, bool)>,
+        current_version_name: Option<&str>,
+        expected: &str,
+    ) {
+        let crate_ = new_crate(
+            versions
+                .into_iter()
+                .map(|(version, yanked)| new_mock_version(version, yanked))
+                .collect(),
+        );
+        let current_version = current_version_name.map(|version_name| {
+            semver::Version::parse(version_name).expect("current_version_name should be valid")
+        });
+        let choosen_baseline = choose_baseline_version(&crate_, current_version.as_ref())
+            .expect("choose_baseline_version should not return any error");
```

Comment:
I think it should be `chosen`, not `choosen`?

Reply:
definitely ;)